src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -442,14 +442,21 @@ protected function build(ContainerBuilder $container)
     /**
      * Gets the container class.
      *
+     * @throws \InvalidArgumentException If the generated classname is invalid
+     *
      * @return string The container class
      */
     protected function getContainerClass()
     {
         $class = \get_class($this);
         $class = 'c' === $class[0] && 0 === strpos($class, "class@anonymous\0") ? get_parent_class($class).str_replace('.', '_', ContainerBuilder::hash($class)) : $class;
+        $class = $this->name.str_replace('\\', '_', $class).ucfirst($this->environment).($this->debug ? 'Debug' : '').'Container';
+
+        if (!preg_match('/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/', $class)) {
+            throw new \InvalidArgumentException(sprintf('The generated classname is invalid. Make sure that the environment name contains only valid characters.'));
```

Comment:
We are trying to make exception message as one sentence. Could be `The generated class name can only contain characters allowed in PHP class names.`

Reply:
Doing it right now.

Reply:
Fixed in c8734c29aec23e66b1123f04645e6edfe66ec77d

Reply:
`Environment "%s" contains invalid characters.`?

Reply:
I think having an explanation about which chars are valid is important. Having the environment name is indeed nice: `Environment "%s" can only contain characters allowed in PHP class names.`

Reply:
Ok will add it.

Reply:
I did a mix of both sentences, I'm waiting for your feedback again.

Reply:
See 1fd718ae1bf370a6d880a6e0d2010c52b3b608d0